### ISSUE TITLE
Conch uses no-longer-present trial benchmark feature

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2159]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2159|
|Type|defect|
|Created|2006-10-08 22:12:09Z|

This means there's some dead code.  Remove it in favor of a standalone benchmark, I guess.

<details><summary>Searchable metadata</summary>

```
trac-id__2159 2159
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__tests tests
time__1160345529000000 1160345529000000
changetime__1162021155000000 1162021155000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

Where is this dead code?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

twisted/conch/test/test_mixin.py

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Ready for review in conch-benchmark-2159

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to @exarkun|
|-|-|

Please add a docstring to buffering_mixin explaining what it is, how it is to be used and why it is in the documentation for conch. If it is meant to be used as a script, then please rename it to something else. Please re-assign to me for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @jml|
|-|-|

Added a module docstring and a readme explaining the purpose of the benchmarks directory.

This stuff is mostly aimed at developers, not end users.  I'm not sure doc/conch/benchmarks/ is the best place, but it is based on the prior art, vis-a-vis doc/core/benchmarks/.

buffering_mixin.py is a runnable script.  To what do you think it should be renamed?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml commented|
|-|-|

`conch-benchmarks` maybe. `benchmark-conch`? `benchmark-conch-buffers`?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to @exarkun|
|-|-|

Merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [18552]) Merge conch-benchmark-2159

Author: exarkun
Reviewer: jml
Fixes [#2159](https://github.com/twisted/twisted/issues/2159)

Move dead benchmark code out of the test suite (where trial used to
run it, but no longer does) and into a dedicated benchmark directory
where it can be run independently.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
